prow/config/config.yaml

```diff
@@ -466,7 +466,8 @@ presubmits:
 
   metal3-io/metal3-dev-env:
   - name: shellcheck
-    always_run: true
+    always_run: false
+    run_if_changed: "(\\.[sh]|^Makefile)$"
```

Comment:
`[sh]` will match strings `s` and `h`, it's like an 'OR'. I think you want parens instead of square brackets here. Also, do you need to double escape the `.` ? I would try: `(\.(sh)|^Makefile)$`.